app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java

```diff
@@ -20,4 +30,56 @@ public static boolean putEventStringsInData(Map<String, String> event, Map<Strin
         }
         return false;
     }
+
+    private static void storeTask(Task task, CollectionReference taskListRef) {
+        Map<String, Object> data = new HashMap<>();
+
+        // Loading information
+        data.put("title", task.getTitle());
+        data.put("description", task.getDescription());
+        data.put("status", task.isFinished() ? "completed" : "ongoing");
+        data.put("owner", task.getOwner().uid());
+
+        List<Map<String, String>> subTaskListData = new ArrayList<>();
+
+        for (Task.SubTask subTask : task.getSubTasks()) {
+            subTaskListData.add(makeSubTaskData(subTask));
+        }
+
+        data.put("sub tasks", subTaskListData);
+
+        taskListRef.add(data);
+    }
+
+    public static Map<String, String> makeSubTaskData(Task.SubTask subTask) {
+        Map<String, String> subTaskData = new HashMap<>();
+        subTaskData.put("title", subTask.getTitle());
+        subTaskData.put("status", subTask.isFinished() ? "completed" : "ongoing");
+
+        return subTaskData;
+    }
+
+    // Might be unnecessary in the future
+    public static void storeTaskList(TaskList taskList, CollectionReference taskListRoot) {
+        Map<String, Object> data = new HashMap<>();
+
+        data.put("title", taskList.getTitle());
+        data.put("owner", taskList.getOwner().uid());
+
+        taskListRoot.add(data).addOnCompleteListener(task -> {
+            if(task.isSuccessful()) {
+                DocumentReference documentReference = task.getResult();
+                CollectionReference taskListRef = documentReference.collection("tasks");
+
+                for (Task t : taskList.getTasks()) {
+                    storeTask(t, taskListRef);
+                }
+            }
+        });
+    }
+
+    public static FirestoreTask recoverTask(Map<String, Object> data, DocumentReference taskDocRef) {
+        return new FirestoreTask(new DummyUser("Recovering-user", (String)data.get("owner")),
+                (String)data.get("title"), (String)data.get("description"), taskDocRef);
+    }
```

Comment:
I don't really understand why these functions are not in FirestoreTask ? Maybe it would be more intuitive to get this kind of functions there

Reply:
It's true that I could put them there. FirestoreTask was meant to be a decorator of Task with added firebase commands, but I suppose I can move these static methods over there too.

Reply:
Maybe the 3rd reviewer can give his opinion so we have a final one ^^